dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -92,19 +87,25 @@ static ClassDumper create(List<ClassPathEntry> entries) throws IOException {
             .collect(toImmutableList());
     checkArgument(
         unreadableFiles.isEmpty(), "Some jar files are not readable: %s", unreadableFiles);
-
-    return new ClassDumper(entries, extensionClassLoader, mapJarToClassFileNames(entries));
```

Comment:
Let me think about that. There might be another way to do this, but it can't involve maintaining multiple multimaps.

I do like the principle of keeping work out of constructors. However I don't believe one should contort the natural design of a class to support testing and mocking. That leads to both bad design and tests that don't actually test the real code. Most of the time more effort spent in test setup avoid the need for mocking.